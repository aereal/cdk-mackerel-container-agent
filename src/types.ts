export enum MackerelContainerPlatform {
  KUBERNETES = "kubernetes", // https://mackerel.io/ja/docs/entry/howto/install-agent/container/kubernetes
  ECS = "ecs" // https://mackerel.io/ja/docs/entry/howto/install-agent/container/ecs
}

export interface ServiceRole {
  role: string;
  service: string;
}

export enum MackerelHostStatus {
  Working = "working",
  Standby = "standby",
  Maintenance = "maintenance",
  Poweroff = "poweroff"
}
