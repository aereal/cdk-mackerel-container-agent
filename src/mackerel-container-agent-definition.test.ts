import { SynthUtils } from "@aws-cdk/assert";
import {
  ContainerImage,
  Ec2TaskDefinition,
  FargateTaskDefinition,
  NetworkMode
} from "@aws-cdk/aws-ecs";
import { Stack } from "@aws-cdk/core";
import {
  MackerelContainerAgentDefinition,
  MackerelContainerAgentImage
} from "./mackerel-container-agent-definition";
import { MackerelHostStatus } from "./types";

describe("MackerelContainerAgentDefinition", () => {
  describe("EC2", () => {
    test("add to task definition with only required props", () => {
      const stack = new Stack();
      const taskDefinition = new Ec2TaskDefinition(stack, "TaskDefinition", {});
      const container = new MackerelContainerAgentDefinition(
        stack,
        "mackerel-container-agent",
        {
          apiKey: "keep-my-secret",
          taskDefinition
        }
      );
      expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
    });

    test("with roles", () => {
      const stack = new Stack();
      const taskDefinition = new Ec2TaskDefinition(stack, "TaskDefinition", {});
      const container = new MackerelContainerAgentDefinition(
        stack,
        "mackerel-container-agent",
        {
          apiKey: "keep-my-secret",
          roles: [
            { service: "My-service", role: "db" },
            { service: "My-service", role: "proxy" }
          ],
          taskDefinition
        }
      );
      expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
    });

    test("specify ignoreContainer", () => {
      const stack = new Stack();
      const taskDefinition = new Ec2TaskDefinition(stack, "TaskDefinition", {});
      const container = new MackerelContainerAgentDefinition(
        stack,
        "mackerel-container-agent",
        {
          apiKey: "keep-my-secret",
          ignoreContainer: "(mackerel|xray)",
          taskDefinition
        }
      );
      expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
    });

    test("with hostStatusOnStart", () => {
      const stack = new Stack();
      const taskDefinition = new Ec2TaskDefinition(stack, "TaskDefinition", {});
      const container = new MackerelContainerAgentDefinition(
        stack,
        "mackerel-container-agent",
        {
          apiKey: "keep-my-secret",
          hostStatusOnStart: MackerelHostStatus.Working,
          taskDefinition
        }
      );
      expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
    });

    test("with customImage", () => {
      const stack = new Stack();
      const taskDefinition = new Ec2TaskDefinition(stack, "TaskDefinition", {});
      const container = new MackerelContainerAgentDefinition(
        stack,
        "mackerel-container-agent",
        {
          apiKey: "keep-my-secret",
          image: ContainerImage.fromRegistry(
            "somebody/some-custom-agent-image"
          ),
          taskDefinition
        }
      );
      expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
    });

    test("with `plugins` image", () => {
      const stack = new Stack();
      const taskDefinition = new Ec2TaskDefinition(stack, "TaskDefinition", {});
      const container = new MackerelContainerAgentDefinition(
        stack,
        "mackerel-container-agent",
        {
          apiKey: "keep-my-secret",
          image: MackerelContainerAgentImage.Plugins,
          taskDefinition
        }
      );
      expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
    });
  });

  describe("EC2 (awsvpc)", () => {
    test("add to task definition", () => {
      const stack = new Stack();
      const taskDefinition = new Ec2TaskDefinition(stack, "TaskDefinition", {
        networkMode: NetworkMode.AWS_VPC
      });
      const container = new MackerelContainerAgentDefinition(
        stack,
        "mackerel-container-agent",
        {
          apiKey: "keep-my-secret",
          taskDefinition
        }
      );
      expect(container.mountPoints).toHaveLength(0);
      expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
    });
  });

  describe("Fargate", () => {
    test("add to task definition", () => {
      const stack = new Stack();
      const taskDefinition = new FargateTaskDefinition(
        stack,
        "TaskDefinition",
        {}
      );
      const container = new MackerelContainerAgentDefinition(
        stack,
        "mackerel-container-agent",
        {
          apiKey: "keep-my-secret",
          taskDefinition
        }
      );
      expect(container.mountPoints).toHaveLength(0);
      expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
    });
  });
});
