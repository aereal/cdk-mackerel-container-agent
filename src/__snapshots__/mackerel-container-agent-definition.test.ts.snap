// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MackerelContainerAgentDefinition EC2 (awsvpc) add to task definition 1`] = `
Object {
  "Resources": Object {
    "TaskDefinitionB36D86D9": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "MACKEREL_CONTAINER_PLATFORM",
                "Value": "ecs",
              },
            ],
            "Essential": true,
            "Image": "mackerel/mackerel-container-agent:latest",
            "Memory": 128,
            "Name": "mackerel-container-agent",
            "Secrets": Array [
              Object {
                "Name": "MACKEREL_APIKEY",
                "ValueFrom": "dummy-arn",
              },
            ],
          },
        ],
        "ExecutionRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionExecutionRole8D61C2FB",
            "Arn",
          ],
        },
        "Family": "TaskDefinition",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": Array [
          "EC2",
        ],
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionTaskRoleFD40A61D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "TaskDefinitionExecutionRole8D61C2FB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "secretsmanager:GetSecretValue",
              "Effect": "Allow",
              "Resource": "dummy-arn",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5",
        "Roles": Array [
          Object {
            "Ref": "TaskDefinitionExecutionRole8D61C2FB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TaskDefinitionTaskRoleFD40A61D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;

exports[`MackerelContainerAgentDefinition EC2 add to task definition with only required props 1`] = `
Object {
  "Resources": Object {
    "TaskDefinitionB36D86D9": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "MACKEREL_CONTAINER_PLATFORM",
                "Value": "ecs",
              },
            ],
            "Essential": true,
            "Image": "mackerel/mackerel-container-agent:latest",
            "Memory": 128,
            "Name": "mackerel-container-agent",
            "Secrets": Array [
              Object {
                "Name": "MACKEREL_APIKEY",
                "ValueFrom": "dummy-arn",
              },
            ],
          },
        ],
        "ExecutionRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionExecutionRole8D61C2FB",
            "Arn",
          ],
        },
        "Family": "TaskDefinition",
        "NetworkMode": "bridge",
        "RequiresCompatibilities": Array [
          "EC2",
        ],
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionTaskRoleFD40A61D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "TaskDefinitionExecutionRole8D61C2FB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "secretsmanager:GetSecretValue",
              "Effect": "Allow",
              "Resource": "dummy-arn",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5",
        "Roles": Array [
          Object {
            "Ref": "TaskDefinitionExecutionRole8D61C2FB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TaskDefinitionTaskRoleFD40A61D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;

exports[`MackerelContainerAgentDefinition EC2 pass unsafeBareAPIKey 1`] = `
Object {
  "Resources": Object {
    "TaskDefinitionB36D86D9": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "MACKEREL_APIKEY",
                "Value": "keep-my-secret",
              },
              Object {
                "Name": "MACKEREL_CONTAINER_PLATFORM",
                "Value": "ecs",
              },
            ],
            "Essential": true,
            "Image": "mackerel/mackerel-container-agent:latest",
            "Memory": 128,
            "Name": "mackerel-container-agent",
            "Secrets": Array [],
          },
        ],
        "Family": "TaskDefinition",
        "NetworkMode": "bridge",
        "RequiresCompatibilities": Array [
          "EC2",
        ],
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionTaskRoleFD40A61D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "TaskDefinitionTaskRoleFD40A61D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;

exports[`MackerelContainerAgentDefinition EC2 specify ignoreContainer 1`] = `
Object {
  "Resources": Object {
    "TaskDefinitionB36D86D9": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "MACKEREL_CONTAINER_PLATFORM",
                "Value": "ecs",
              },
              Object {
                "Name": "MACKEREL_IGNORE_CONTAINER",
                "Value": "(mackerel|xray)",
              },
            ],
            "Essential": true,
            "Image": "mackerel/mackerel-container-agent:latest",
            "Memory": 128,
            "Name": "mackerel-container-agent",
            "Secrets": Array [
              Object {
                "Name": "MACKEREL_APIKEY",
                "ValueFrom": "dummy-arn",
              },
            ],
          },
        ],
        "ExecutionRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionExecutionRole8D61C2FB",
            "Arn",
          ],
        },
        "Family": "TaskDefinition",
        "NetworkMode": "bridge",
        "RequiresCompatibilities": Array [
          "EC2",
        ],
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionTaskRoleFD40A61D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "TaskDefinitionExecutionRole8D61C2FB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "secretsmanager:GetSecretValue",
              "Effect": "Allow",
              "Resource": "dummy-arn",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5",
        "Roles": Array [
          Object {
            "Ref": "TaskDefinitionExecutionRole8D61C2FB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TaskDefinitionTaskRoleFD40A61D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;

exports[`MackerelContainerAgentDefinition EC2 with \`plugins\` image 1`] = `
Object {
  "Resources": Object {
    "TaskDefinitionB36D86D9": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "MACKEREL_CONTAINER_PLATFORM",
                "Value": "ecs",
              },
            ],
            "Essential": true,
            "Image": "mackerel/mackerel-container-agent:plugins",
            "Memory": 128,
            "Name": "mackerel-container-agent",
            "Secrets": Array [
              Object {
                "Name": "MACKEREL_APIKEY",
                "ValueFrom": "dummy-arn",
              },
            ],
          },
        ],
        "ExecutionRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionExecutionRole8D61C2FB",
            "Arn",
          ],
        },
        "Family": "TaskDefinition",
        "NetworkMode": "bridge",
        "RequiresCompatibilities": Array [
          "EC2",
        ],
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionTaskRoleFD40A61D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "TaskDefinitionExecutionRole8D61C2FB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "secretsmanager:GetSecretValue",
              "Effect": "Allow",
              "Resource": "dummy-arn",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5",
        "Roles": Array [
          Object {
            "Ref": "TaskDefinitionExecutionRole8D61C2FB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TaskDefinitionTaskRoleFD40A61D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;

exports[`MackerelContainerAgentDefinition EC2 with customImage 1`] = `
Object {
  "Resources": Object {
    "TaskDefinitionB36D86D9": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "MACKEREL_CONTAINER_PLATFORM",
                "Value": "ecs",
              },
            ],
            "Essential": true,
            "Image": "somebody/some-custom-agent-image",
            "Memory": 128,
            "Name": "mackerel-container-agent",
            "Secrets": Array [
              Object {
                "Name": "MACKEREL_APIKEY",
                "ValueFrom": "dummy-arn",
              },
            ],
          },
        ],
        "ExecutionRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionExecutionRole8D61C2FB",
            "Arn",
          ],
        },
        "Family": "TaskDefinition",
        "NetworkMode": "bridge",
        "RequiresCompatibilities": Array [
          "EC2",
        ],
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionTaskRoleFD40A61D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "TaskDefinitionExecutionRole8D61C2FB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "secretsmanager:GetSecretValue",
              "Effect": "Allow",
              "Resource": "dummy-arn",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5",
        "Roles": Array [
          Object {
            "Ref": "TaskDefinitionExecutionRole8D61C2FB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TaskDefinitionTaskRoleFD40A61D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;

exports[`MackerelContainerAgentDefinition EC2 with hostStatusOnStart 1`] = `
Object {
  "Resources": Object {
    "TaskDefinitionB36D86D9": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "MACKEREL_CONTAINER_PLATFORM",
                "Value": "ecs",
              },
              Object {
                "Name": "MACKEREL_HOST_STATUS_ON_START",
                "Value": "working",
              },
            ],
            "Essential": true,
            "Image": "mackerel/mackerel-container-agent:latest",
            "Memory": 128,
            "Name": "mackerel-container-agent",
            "Secrets": Array [
              Object {
                "Name": "MACKEREL_APIKEY",
                "ValueFrom": "dummy-arn",
              },
            ],
          },
        ],
        "ExecutionRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionExecutionRole8D61C2FB",
            "Arn",
          ],
        },
        "Family": "TaskDefinition",
        "NetworkMode": "bridge",
        "RequiresCompatibilities": Array [
          "EC2",
        ],
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionTaskRoleFD40A61D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "TaskDefinitionExecutionRole8D61C2FB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "secretsmanager:GetSecretValue",
              "Effect": "Allow",
              "Resource": "dummy-arn",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5",
        "Roles": Array [
          Object {
            "Ref": "TaskDefinitionExecutionRole8D61C2FB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TaskDefinitionTaskRoleFD40A61D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;

exports[`MackerelContainerAgentDefinition EC2 with roles 1`] = `
Object {
  "Resources": Object {
    "TaskDefinitionB36D86D9": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "MACKEREL_CONTAINER_PLATFORM",
                "Value": "ecs",
              },
              Object {
                "Name": "MACKEREL_ROLES",
                "Value": "My-service:db,My-service:proxy",
              },
            ],
            "Essential": true,
            "Image": "mackerel/mackerel-container-agent:latest",
            "Memory": 128,
            "Name": "mackerel-container-agent",
            "Secrets": Array [
              Object {
                "Name": "MACKEREL_APIKEY",
                "ValueFrom": "dummy-arn",
              },
            ],
          },
        ],
        "ExecutionRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionExecutionRole8D61C2FB",
            "Arn",
          ],
        },
        "Family": "TaskDefinition",
        "NetworkMode": "bridge",
        "RequiresCompatibilities": Array [
          "EC2",
        ],
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionTaskRoleFD40A61D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "TaskDefinitionExecutionRole8D61C2FB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "secretsmanager:GetSecretValue",
              "Effect": "Allow",
              "Resource": "dummy-arn",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5",
        "Roles": Array [
          Object {
            "Ref": "TaskDefinitionExecutionRole8D61C2FB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TaskDefinitionTaskRoleFD40A61D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;

exports[`MackerelContainerAgentDefinition Fargate add to task definition 1`] = `
Object {
  "Resources": Object {
    "TaskDefinitionB36D86D9": Object {
      "Properties": Object {
        "ContainerDefinitions": Array [
          Object {
            "Environment": Array [
              Object {
                "Name": "MACKEREL_CONTAINER_PLATFORM",
                "Value": "ecs",
              },
            ],
            "Essential": true,
            "Image": "mackerel/mackerel-container-agent:latest",
            "Memory": 128,
            "Name": "mackerel-container-agent",
            "Secrets": Array [
              Object {
                "Name": "MACKEREL_APIKEY",
                "ValueFrom": "dummy-arn",
              },
            ],
          },
        ],
        "Cpu": "256",
        "ExecutionRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionExecutionRole8D61C2FB",
            "Arn",
          ],
        },
        "Family": "TaskDefinition",
        "Memory": "512",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": Array [
          "FARGATE",
        ],
        "TaskRoleArn": Object {
          "Fn::GetAtt": Array [
            "TaskDefinitionTaskRoleFD40A61D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "TaskDefinitionExecutionRole8D61C2FB": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
    "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "secretsmanager:GetSecretValue",
              "Effect": "Allow",
              "Resource": "dummy-arn",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "TaskDefinitionExecutionRoleDefaultPolicy1F3406F5",
        "Roles": Array [
          Object {
            "Ref": "TaskDefinitionExecutionRole8D61C2FB",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TaskDefinitionTaskRoleFD40A61D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "ecs-tasks.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
